metrics/prometheus/prometheus.go

```diff
@@ -118,3 +118,24 @@ func (r recorder) ObserveHTTPResponseSize(_ context.Context, id string, sizeByte
 func (r recorder) AddInflightRequests(_ context.Context, id string, quantity int) {
 	r.httpRequestsInflight.WithLabelValues(id).Add(float64(quantity))
 }
+
+// RegisterHTTPRequestDurationValues allows for pre-registering values of id,
+// duration, method and code so that the coresponding metrics are emited
+// right from the start of the recorder and not just with the first request
+func (r recorder) RegisterHTTPRequestDurationValues(id string, method, code string) {
+	r.httpRequestDurHistogram.WithLabelValues(id, method, code)
```

Comment:
Calling `.WithLabelValues` on a histogram is enough to make Prometheus client register the new metric.